jobManager.py

```diff
@@ -63,63 +63,53 @@ def _getNextID(self):
     def __manage(self):
         self.running = True
         while True:
-            id = self.jobQueue.getNextPendingJob()
-
-            if id:
-                job = self.jobQueue.get(id)
-
-                # job could no longer exist if it was completed by someone else
-                if job == None:
-                    continue
-
-                if not job.accessKey and Config.REUSE_VMS:
-                    id, vm = self.jobQueue.getNextPendingJobReuse(id)
-                    job = self.jobQueue.get(id)
-                    if job == None:
-                        continue
-
-                try:
-                    # Mark the job assigned
-                    self.jobQueue.assignJob(job.id)
-                    # if the job has specified an account
-                    # create an VM on the account and run on that instance
-                    if job.accessKeyId:
-                        from vmms.ec2SSH import Ec2SSH
-                        vmms = Ec2SSH(job.accessKeyId, job.accessKey)
-                        newVM = copy.deepcopy(job.vm)
-                        newVM.id = self._getNextID()
-                        preVM = vmms.initializeVM(newVM)
+            # Blocks until we get a next job
+            job = self.jobQueue.getNextPendingJob()
+
+            if not job.accessKey and Config.REUSE_VMS:
+                vm = None
+                while (vm is None):
```

Comment:
Probably want to introduce some delay here

Reply:
Also just a nit, but the parenthesis around the condition is unnecessary

Reply:
Actually this was probably an unintended good catch. I shifted the sleep into this while loop

Reply:
since the getPendingJob now blocks until the next job available, there is no need to sleep in the outer loop